api/src/main/java/com/velocitypowered/api/proxy/Player.java

```diff
@@ -218,5 +218,21 @@ default void sendMessage(net.kyori.text.Component component) {
    * @param url the URL for the resource pack
    * @param hash the SHA-1 hash value for the resource pack
    */
-  void sendResourcePack(String url, byte[] hash);
+  default void sendResourcePack(String url, byte[] hash) {
+    sendResourcePack(url, hash, false);
+  }
+
+  /**
+   * Sends the specified resource pack from {@code url} to the user, using the specified 20-byte
+   * SHA-1 hash. To monitor the status of the sent resource pack, subscribe to
+   * {@link PlayerResourcePackStatusEvent}.
+   * In 1.17 and newer you can additionally specify
+   * whether the resource pack is required or not. Setting this for an older client will have
+   * no effect.
+   *
+   * @param url the URL for the resource pack
+   * @param hash the SHA-1 hash value for the resource pack
+   * @param isRequired flag to set the resource pack as required in 1.17+
```

Comment:
Could you specify in the JD what is meant by 'isRequired'? It's unclear at first glance; I had to check wiki.vg

Forced | Boolean | The notchian client will be forced to use the resource pack from the server. If they decline they will be kicked from the server

Reply:
Point taken. The final description will be shorter to the sum of: isRequired refusing this pack on 1.17 or newer disconnects the user.